src/skmultiflow/meta/streaming_random_patches.py

```diff
@@ -0,0 +1,667 @@
+                                          n_samples_seen=n_samples_seen,
+                                          random_state=random_state)
+
+            # ===== Drift detection =====
+            # Update the drift detection method
+            self.drift_detection_method.add_element(0 if correctly_classifies else 1)
+            # Check if the was a change
+            if self.drift_detection_method.detected_change():
+                self.n_drifts_detected += 1
+                # There was a change, reset the model
+                self.reset(n_features=n_features_total, n_samples_seen=n_samples_seen,
+                           random_state=random_state)
+
+    def predict_proba(self, X):
+        if self.feature_indexes is not None:
+            # Select the subset of features to use
+            X_subset = np.asarray([X[0][self.feature_indexes]])
+        else:
+            # Use all features
+            X_subset = X
+
+        return self.base_estimator.predict_proba(X_subset)
+
+    def _trigger_warning(self, n_features, n_samples_seen: int, random_state: np.random):
+        background_base_estimator = clone(self.base_estimator)
+        background_base_estimator.reset()
+
+        background_performance_evaluator = deepcopy(self.performance_evaluator)
+        background_performance_evaluator.reset()
+
+        feature_indexes = self._reset_subset(n_features=n_features, random_state=random_state)
+
+        self._background_learner = self._background_learner_class(
+            idx_original=self.idx_original,
+            base_estimator=background_base_estimator,
+            performance_evaluator=background_performance_evaluator,
+            created_on=n_samples_seen,
+            disable_background_learner=self.disable_background_learner,
+            disable_drift_detector=self.disable_drift_detector,
+            drift_detection_method=self.drift_detection_method,
+            warning_detection_method=self.warning_detection_method,
+            drift_detection_criteria=self.drift_detection_criteria,
+            is_background_learner=True,
+            feature_indexes=feature_indexes,
+            nominal_attributes=self.nominal_attributes,
+            random_state=self._random_state
+        )
+
+        # Hard-reset the warning method
+        self.warning_detection_method = clone(self.warning_detection_method)
+
+    def _reset_subset(self, n_features: int, random_state: np.random):
+        feature_indexes = None
+        if self.feature_indexes is not None:
+            k = len(self.feature_indexes)
+            feature_indexes = random_state.choice(range(n_features), k, replace=False)
+        return feature_indexes
+
+    def reset(self, n_features: int, n_samples_seen: int, random_state: np.random):
+        if not self.disable_background_learner and self._background_learner is not None:
+            self.base_estimator = deepcopy(self._background_learner.base_estimator)
```

Comment:
I this case I believe that we could simply assign `self._background_learner.base_estimator` to `self.base_estimator`. What do you think? Is there a specific reason to copy the whole model?

Reply:
You are right. Removed it in 429b21a12408949d9bcb24ac8bb3e4e577f97224